trace/trace.go

```diff
@@ -472,3 +577,11 @@ func (s *Span) createChildSpan(ctx context.Context, async bool) (context.Context
 	ctx = PutSpanInContext(ctx, newSpan)
 	return ctx, newSpan
 }
+
+// generateID generates random bytes and encodes them as a hex string. The
+// length of the id depends on the length of the byte slice provided. If
+// io.ReadAll fails, the result will be all zeroes.
```

Comment:
`io.ReadFull`, not `io.ReadAll` - also, is all zeros the best we can do on failure - not that this should ever happen - although running out of entropy is a thing? I suppose there aren't a lot of better alternatives - although maybe even an incrementing counter might be better than all the IDs being zero.
Also, in researching, you can just use `rand.Read` here: https://golang.org/pkg/crypto/rand/#Read

Reply:
Thanks for the catch and info re: `rand.Read`. That's much better. 

I went back and forth a bit on the defaults and decided that yeah, it's nice to make a best effort in the case that we can't generate a pseudo-random ID. I've updated the function to use a counter as the fallback as you suggest.